rocky/rocky/views/ooi_tree.py

```diff
@@ -17,20 +14,27 @@ class OOITreeView(BaseOOIDetailView):
     template_name = "oois/ooi_tree.html"
     connector_form_class = OoiTreeSettingsForm
 
-    def get_tree_dict(self):
-        return create_object_tree_item_from_ref(self.tree.root, self.tree.store)
-
-    def get_filtered_tree(self, tree_dict):
-        filtered_types = self.request.GET.getlist("ooi_type", [])
-        return filter_ooi_tree(tree_dict, filtered_types)
+    def setup(self, request, *args, **kwargs):
+        super().setup(request, *args, **kwargs)
+        self.ooi_id = self.request.GET.get("ooi_id", "")
+        self.observed_at = self.get_observed_at()
+        self.depth = self.get_depth()
+        self.filtered_types = self.request.GET.getlist("ooi_type", [])
+        self.view = self.request.GET.get("view", "condensed")
+
+        try:
+            self.tree = self.get_ooi_tree(self.ooi_id, self.depth, self.observed_at)
+            self.tree_objects = create_object_tree_item_from_ref(self.tree.root, self.tree.store)
+            self.ooi_types = get_ooi_types_from_tree(self.tree_objects, False)
+            self.filtered_tree_objects = filter_ooi_tree(self.tree_objects, self.filtered_types)
+            self.hydrated_tree = hydrate_tree(self.filtered_tree_objects, self.organization.code)
+        except HTTPError:
+            messages.error(request, _("We could not process your request."))
```

Comment:
Why did you move all the code to `setup`? `setup` method is supposed to be used for view initialization (see https://docs.djangoproject.com/en/4.2/ref/class-based-views/base/#django.views.generic.base.View.setup), I don't think it is a good pattern to move most of the view code to `setup`.

If I understand things correctly you also ignore the things already done in `BaseOOIDetailView`, I think this is what causes it to fetch data twice.

The same problem here with catching the exception and continuing. The attributes won't be set and we will get an attribute error later on.

We also shouldn't move the hydration code from `OOIGraphView` to here if it is only needed for `OOIGraphView`, because that means `OOISummaryView` will do more work than necessary.

Reply:
I put all back and make this PR a bit smaller, see all new changes